akka-docs/src/test/java/jdocs/stream/operators/sourceorflow/Monitor.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.stream.operators.sourceorflow;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.japi.Pair;
+import akka.stream.FlowMonitor;
+import akka.stream.FlowMonitorState;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/** */
+public class Monitor {
+
+  // #monitor
+  private static <T> void printMonitorState(FlowMonitorState.StreamState<T> state) {
+    if (FlowMonitorState.Finished$.class.isInstance(state)) {
```

Comment:
Java api to access singleton values is actually available, so

`if (state == FlowMonitorState.finished())` 

Also for received and failed which will be instances I would have expected Java style instanceof keyword rather than method calls:

`if (state instanceof FlowMonitorState.Received)`